qiskit/compiler/assembler.py

```diff
@@ -321,7 +322,7 @@ def _parse_common_args(
             shots = min(1024, max_shots)
         else:
             shots = 1024
-    elif not isinstance(shots, int):
+    elif not isinstance(shots, numbers.Integral):
```

Comment:
I think it would be better to use `numpy.integer` here instead of the abstract numbers class from stdlib. We've tried to  use the abstract `numbers` classes before and had performance regressions each time (because of how the abstract classes hook into the checks for isinstance). While assemble isn't super performance critical it's still used in job submission path a surprising number of place and this is only just a single `isinstance` call so the extra overhead isn't as huge as something checked on each instruction in a circuit, I still think it would probably be better to just use `numpy.integer` even if that means other `int` types from other external libraries can't be used. In practice I don't think anything besides numpy ints will really come up.

Reply:
Sounds reasonable; I've pushed a fix.